eng/pipelines/templates/steps/analyze.yml

```diff
@@ -4,18 +4,60 @@ parameters:
     default: stable
 
 steps:
+  - ${{ if ne(parameters.Toolchain, 'nightly') }}:
+      - template: /eng/pipelines/templates/steps/use-rust.yml@self
+        parameters:
+          Toolchain: nightly
+
   - template: /eng/pipelines/templates/steps/use-rust.yml@self
     parameters:
       Toolchain: ${{ parameters.Toolchain }}
 
+  - template: /eng/common/pipelines/templates/steps/save-package-properties.yml
+
   - pwsh: |
+      . ./eng/common/scripts/common.ps1
```

Comment:
We should go ahead and move this into a script. It was already getting large but this pushes it over for me.

Reply:
moved to script file